common/djangoapps/student/views/dashboard.py

```diff
@@ -767,6 +767,8 @@ def student_dashboard(request):
         redirect_message = _("The course you are looking for is closed for enrollment as of {date}.").format(
             date=request.GET['course_closed']
         )
+    elif 'expired_message' in request.GET:
+        redirect_message = request.GET['expired_message']
```

Comment:
This is moving to a pattern of just displaying the message directly from the error... Maybe we should make it clear that it's a generic message, so that other errors could be moved to that? (For instance, the other two types of error messages above could just be pushed down into `AccessResponse` objects...)